apluslms_roman/cli.py

```diff
@@ -379,6 +413,142 @@ def config_print_action(context):
+        # so we need to check if the file actually exists
+        if not isfile(config_path):
+            raise RomanConfigError
+        exit(1, "{} already exists in {}.\nNo new Roman config created.".format(
+            basename(config_path), cwd)
+        )
+    except RomanConfigError:
+        filename = DEFAULT_NAME + ".yml"
+        config = LEGACY_CONFIG if is_legacy_configuration(cwd) else DEFAULT_CONFIG
+        filepath = path_join(cwd, filename)
+        with open(filepath, 'w') as outfile:
+            yaml_dump(config, outfile)
+        print("{} successfully created in {}".format(filename, cwd))
+
+def set_args(context):
+    if context.args.user == context.args.project:
+        raise Exception("Choose either user settings or project settings")
+    if context.args.user:
+        document = context.settings
+    else:
+        document = context.config
+
+    schema = document.validator.schema
+    for val in context.args.values:
+        try:
+            key, val = val.split('=')
+            parts = key.split(".")
+            val_type = schema
+            try:
+                for part in parts:
+                    val_type = val_type["properties"][part]
+                val_type = val_type["type"]
+                if val_type == "integer":
+                    if val.isdigit():
+                        val = int(val)
+                    else:
+                        raise TypeError("'" + key + "' should be an integer.")
+                elif val_type == "boolean":
+                    if val.lower == "true" or val == "1":
+                        val = True
+                    elif val.lower == "false" or val == "0":
+                        val = False
+                    else:
+                        raise TypeError("'" + key + "' should be a boolean.")
+            except KeyError:
+                raise KeyError("'" + key + "' is not a key found in the schema.")
+            document.mlset(key, val)
+        except ValueError:
+            raise ValueError("Give values in format 'key=val'.")
+    document.validate()
+    document.save()
+
+def del_args(context):
+    if context.args.user == context.args.project:
+        raise Exception("Choose either user settings or project settings")
+    if context.args.user:
+        document = context.settings
+    else:
+        document = context.config
+    for val in context.args.keys:
+        document.mldel(val)
+    document.validate()
+    document.save()
+
+def add_step(context):
+    args = context.args
+    env = args.env
+    if env:
+        try:
+            env = {a: b for a, b in [s.split('=') for s in env]}
+        except ValueError:
+            raise ValueError("env is a dict, so values need to be in key=val format, e.g. a=1 b=2")
+    step = {
+        "img": args.img,
+        "cmd": args.cmd,
+        "mnt": args.mnt,
+        "env": env,
+        "name": args.name
+    }
+    nulls = [k for k in step if not step[k]]
```

Comment:
`step = {k: v for k, v in step.items() if v}` is probably simpler than removing empties.